Android Auto Connectivity Broken

**Only issues related to the code of The Open GApps Project scripts themselves**

If you have found an issue with a pre-built OpenGApps.org package please use the XDA-Developers [forum thread](http://forum.xda-developers.com/android/software/Open-GApps-t3098071)

**Checklist**

- [X] The issue is not answered by reading the [FAQ](https://github.com/opengapps/opengapps/wiki/FAQ)
- [X] The issue is part of The Open GApps Project scope *(and not of the pre-built OpenGApps.org packages scope, for support on those see [XDA Forum](http://forum.xda-developers.com/android/software/Open-GApps-t3098071))*

*If Open GApps compilation fails:*
- [X] The issue is not a [duplicate](https://github.com/opengapps/opengapps/issues?q=is%3Aissue)

*If Open GApps installation fails:*
- [X] Is the issue reproducible using the latest version of your recovery?
- [X] Is the issue reproducible with (the latest) [TWRP](https://twrp.me) recovery? *(other recoveries are only best-effort)*

*If receiving force closures after Open GApps installation:*
- [X] The issue is **NOT** an error within one of the bundled Google applications itself *(verify by e.g. installing the application from the Play Store instead)*
- [X] The issue is reproducible on clean (=wiped ``/data/`` partition) installation
- [X] The affected ROM includes the necessary patches and functionality as noted for [Android versions](https://github.com/opengapps/opengapps/wiki#notes-for-android-versions) and [Platforms](https://github.com/opengapps/opengapps/wiki#notes-for-platforms)
- [X] No non-AOSP hacks like privacy guards' permission restrictions etcetera are applied

**Issue description:**
When I plug my phone into my car's Android Auto dash it doesn't work and says "Please unplug, wait a moment and re-plug your device." Running LineageOS 16 for Payton and opengapps (latest builds). Works fine with a rooted stock build.

**Reproduction Steps:**

1. Install lineage + twrp + opengapps
2. Boot phone, complete setup, install Android Auto and Google Maps
3. Connect phone to automobile with wire
4. Android Auto crashes

**Expected behavior:**

Connecting phone to car's dashboard should launch android auto instead of an error.

**Observed behavior:**

Android Auto won't work with OpenGApps and MindTheGApps regardless of what rom is used (I've tested Oreo and Pie roms). It will work just fine with stock device builds. It is not  because the cable is bad. Android Auto application launches just fine on the device, but it does not work tethered to an automobile's dashboard.




[2019-05-26-21-02-00.zip](https://github.com/opengapps/opengapps/files/3235084/2019-05-26-21-02-00.zip)

I got issues on Android 11 with Android Auto installed from Playstore with OpenGapps Nano.
I think the way permissions are accquired in later Android versions are requiring some settings in either packages.xml
or i need to manually flash it as priv-app or something. But whenever i manually edit these files, they get resetted.

`[11-13 15:03:12.273 12148:12148 E/AndroidRuntime]
FATAL EXCEPTION: main
Process: com.google.android.projection.gearhead:car, PID: 12148
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.android.projection.gearhead/com.google.android.apps.auto.carservice.gmscorecompat.FirstActivityImpl}: java.lang.SecurityException: Neither user 10299 nor current process has android.permission.MANAGE_USB.
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7656)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.SecurityException: Neither user 10299 nor current process has android.permission.MANAGE_USB.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2373)
	at android.os.Parcel.createException(Parcel.java:2357)
	at android.os.Parcel.readException(Parcel.java:2340)
	at android.os.Parcel.readException(Parcel.java:2282)
	at android.hardware.usb.IUsbManager$Stub$Proxy.getPorts(IUsbManager.java:1706)
	at android.hardware.usb.UsbManager.getPorts(UsbManager.java:875)
	at com.google.android.gms.car.usb.SystemUsbStatusProvider.a(SourceFile:4)
	at knu.a(SourceFile:9)
	at knu.a(SourceFile:51)
	at kmu.a(SourceFile:31)
	at com.google.android.gms.carsetup.FirstActivityImpl.onCreate(SourceFile:89)
	at android.app.Activity.performCreate(Activity.java:8000)
	at android.app.Activity.performCreate(Activity.java:7984)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
	... 11 more
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.ContextImpl.enforce(ContextImpl.java:2018)
	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2046)
	at com.android.server.usb.UsbService.getPorts(UsbService.java:655)
	at android.hardware.usb.IUsbManager$Stub.onTransact(IUsbManager.java:828)
	at android.os.Binder.execTransactInternal(Binder.java:1154)
`

Got this error on `stock` arm64 GApps from 5 December 2020.

```
12-05 22:43:43.623 19267 19291 E GmsClient: unable to connect to service: com.google.android.gms.car.service.START on com.google.android.gms
12-05 22:43:43.630 19267 19287 E GH.CAR  : getCarService failed
12-05 22:43:43.630 19267 19287 E GH.CAR  : java.lang.IllegalStateException: Failed to get the delegate car service.
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at byn.k(SourceFile:25)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at cco.k(SourceFile:2)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at cbr.a(Unknown Source:5)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at com.google.android.apps.auto.carservice.service.impl.GearheadCarStartupService.e(SourceFile:1)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at cco.j(SourceFile:2)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at com.google.android.gms.car.startup.IStartup$Stub.dispatchTransaction(SourceFile:69)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at brw.onTransact(SourceFile:2)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at android.os.Binder.execTransactInternal(Binder.java:1021)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at android.os.Binder.execTransact(Binder.java:994)
12-05 22:43:43.630 19267 19287 E GH.CAR  : Caused by: com.google.android.gms.common.api.ApiException: 17: API: Car.API is not available on this device. Connection failed with: ConnectionResult{statusCode=API_UNAVAILABLE, resolution=null, message=null}
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at lpm.c(SourceFile:1)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.s(SourceFile:7)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.k(SourceFile:2)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.f(SourceFile:21)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.dM(SourceFile:1)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at lqw.run(SourceFile:10)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at android.os.Handler.handleCallback(Handler.java:883)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at android.os.Handler.dispatchMessage(Handler.java:100)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at android.os.Looper.loop(Looper.java:214)
12-05 22:43:43.630 19267 19287 E GH.CAR  : 	at android.os.HandlerThread.run(HandlerThread.java:67)
12-05 22:43:43.631 19267 19481 E GH.CAR  : getCarService failed
12-05 22:43:43.631 19267 19481 E GH.CAR  : java.lang.IllegalStateException: Failed to get the delegate car service.
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at byn.k(SourceFile:25)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at cco.k(SourceFile:2)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at cbr.a(Unknown Source:5)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at com.google.android.apps.auto.carservice.service.impl.GearheadCarStartupService.e(SourceFile:1)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at cco.j(SourceFile:2)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at com.google.android.gms.car.startup.IStartup$Stub.dispatchTransaction(SourceFile:69)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at brw.onTransact(SourceFile:2)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at android.os.Binder.execTransactInternal(Binder.java:1021)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at android.os.Binder.execTransact(Binder.java:994)
12-05 22:43:43.631 19267 19481 E GH.CAR  : Caused by: com.google.android.gms.common.api.ApiException: 17: API: Car.API is not available on this device. Connection failed with: ConnectionResult{statusCode=API_UNAVAILABLE, resolution=null, message=null}
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at lpm.c(SourceFile:1)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.s(SourceFile:7)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.k(SourceFile:2)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.f(SourceFile:21)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at com.google.android.gms.common.api.internal.GoogleApiManager$ClientConnection.dM(SourceFile:1)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at lqw.run(SourceFile:10)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at android.os.Handler.handleCallback(Handler.java:883)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at android.os.Handler.dispatchMessage(Handler.java:100)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at android.os.Looper.loop(Looper.java:214)
12-05 22:43:43.631 19267 19481 E GH.CAR  : 	at android.os.HandlerThread.run(HandlerThread.java:67)
12-05 22:43:43.644 19267 19291 E GmsClient: unable to connect to service: com.google.android.gms.car.service.START on com.google.android.gms
```

--- a/scripts/inc.buildtarget.sh
+++ b/scripts/inc.buildtarget.sh
@@ -212,7 +212,7 @@ get_package_info(){
     webviewgoogle)            packagetype="GApps"; packagename="com.google.android.webview"; packagetarget="app/WebViewGoogle"; packagegappsremove="$webviewgappsremove";;
     trichromelibrary)         packagetype="GApps"; packagename="com.google.android.trichromelibrary"; packagetarget="app/TrichromeLibrary";;
     webviewstub)              packagetype="GApps"; packagename="com.google.android.webview.stub"; packagetarget="app/WebViewStub";;
-    gearheadstub)             packagetype="GApps"; packagename="com.google.android.projection.gearhead.stub"; packagetarget="app/AndroidAutoPrebuiltStub";;
+    gearheadstub)             packagetype="GApps"; packagename="com.google.android.projection.gearhead.stub"; packagetarget="priv-app/AndroidAutoPrebuiltStub";;
 
     # Regular GApps
     backuprestore)            packagetype="Core"; packagename="com.google.android.apps.restore"; packagetarget="priv-app/GoogleRestore";;

